### ISSUE TITLE
Lady vashj (BB #799)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#799]
This issue was migrated from bitbucket.
**Original Reporter:** Areradon
**Original Date:** 29.01.2015 10:13:06 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** new
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/799

<hr>

Phase 1: When Vashj roots the player she should run away (getmovementmaster)and fire multi-shot, this would need to be cased

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 29.01.2015 16:18:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/799/#comment-15333511

<hr>

Issue #798 was marked as a duplicate of this issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Areradon
**Original Date:** 29.01.2015 20:02:18 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/799/#comment-15338454

<hr>

Oh sorry, i was posting this issue with my handy :D

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Areradon
**Original Date:** 01.03.2015 11:34:38 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/799/#comment-16009533

<hr>

```
#!c++

@@ -170,6 +170,7 @@
     bool InCombat;
     bool Intro;
     bool CanAttack;
+   bool AfterMoving;

     void Reset()
     {
@@ -192,7 +193,9 @@
         Intro = false;

         me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE); //set it only once on creature create (no need do intro if wiped)
+       m_creature->GetMotionMaster()->MoveIdle();

+       AfterMoving = false;
         Entangle = false;
         CanAttack = false;

@@ -311,6 +314,49 @@
         }
     }

+   void MovementInform(uint32 type, uint32 id)
+    {
+        if(type == POINT_MOTION_TYPE)
+        {
+           me->GetMotionMaster()->Clear();
+           Check_Timer = 1000;
+            AfterMoving = true;
+        }
+    }
+
+   void MoveToOuterEdge()
+   {
+       switch (urand(0, 8))
+        {
+        case 0:
+            me->GetMotionMaster()->MovePoint(0, 7.47564, -910.011, 42.0125);
+            break;
+        case 1:
+            me->GetMotionMaster()->MovePoint(0, 16.9609, -901.763, 42.3111);
+            break;
+        case 2:
+            me->GetMotionMaster()->MovePoint(0, 29.6374, -897.657, 42.1208);
+            break;
+       case 3:
+           me->GetMotionMaster()->MovePoint(0, 42.9139, -901.117, 42.064);
+           break;
+       case 4:
+           me->GetMotionMaster()->MovePoint(0, 51.8512, -910.503, 42.1712);
+           break;
+       case 5:
+           me->GetMotionMaster()->MovePoint(0, 55.089, -923.342, 42.2608);
+           break;
+       case 6:
+           me->GetMotionMaster()->MovePoint(0, 52.0643, -936.46, 42.088);
+           break;
+       case 7:
+           me->GetMotionMaster()->MovePoint(0, 42.6871, -945.147, 42.2546);
+           break;
+       case 8:
+           me->GetMotionMaster()->MovePoint(0, 29.6826, -948.737, 42.2689);
+           break;
+       }
+    }
     void CastShootOrMultishot()
     {
         //Shoot
@@ -375,6 +421,12 @@
                ParalyzeCheck_Timer -= diff;
         }

+       if(AfterMoving)
+       {
+           m_creature->GetMotionMaster()->MoveIdle();
+           AfterMoving = false;
+       }
+
         if(Phase == 1 || Phase == 3)
         {
             //ShockBlast_Timer
@@ -413,7 +465,9 @@
                 {
                     //Entangle
                     //Used in Phases 1 and 3, it casts Entangling Roots on everybody in a 15 yard radius of Vashj, immobilzing them for 10 seconds and dealing 500 damage every 2 seconds. It's not a magic effect so it cannot be dispelled, but is removed by various buffs such as Cloak of Shadows or Blessing of Freedom.
-                    DoCast(me->getVictim(), SPELL_ENTANGLE);
+                    m_creature->AttackStop();
+                   DoCast(me->getVictim(), SPELL_ENTANGLE);
+                   MoveToOuterEdge();
                     Entangle = true;
                     Entangle_Timer = 10000;
                 }
@@ -421,6 +475,7 @@
                 {
                     CastShootOrMultishot();
                     Entangle = false;
+                   Check_Timer = 3000;
                     Entangle_Timer = 20000+rand()%5000;
                 }
             }
@@ -502,7 +557,14 @@

                 //if nobody is in melee range
                 if(!InMeleeRange)
-                    CastShootOrMultishot();
+               {
+                    CastShootOrMultishot();    
+                   m_creature->GetMotionMaster()->MoveIdle();
+               }
+               else if (InMeleeRange)
+               {
+                   me->GetMotionMaster()->MoveChase(me->getVictim());
+               }

                 Check_Timer = 2000;
             }else Check_Timer -= diff;

```

I have done this now. but her dont stop moving when her reach the waypoint.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 21.04.2015 16:19:26 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/799/#comment-17553141

<hr>

You should use combatmovement is built into script ai
